include/deal.II/base/smartpointer.h

```diff
@@ -396,6 +423,31 @@ SmartPointer<T, P>::operator=(const SmartPointer<T, P> &tt)
 
 
 
+template <typename T, typename P>
+inline SmartPointer<T, P> &
+SmartPointer<T, P>::operator=(SmartPointer<T, P> &&tt) noexcept
+{
+  // if objects on the left and right hand side of the operator= are
+  // the same, then this is a no-op
+  if (&tt != this)
+    {
+      // Let us unsubscribe from the current object
+      if (pointed_to_object_is_alive && t != nullptr)
+        t->unsubscribe(&pointed_to_object_is_alive, id);
+
+      // Then reset to the new object, and subscribe to it
+      t = tt.get();
```

Comment:
get() fails if tt is empty. Do you want to allow that situation? Also the if in the line below doesn't make much sense, right?

Reply:
Yes, good point. So fixed.